File: MyApplication/MyApplication/app/src/main/java/com/example/myapplication/api/ApiService.java
package com.example.myapplication.api;

import com.example.myapplication.model.Ledger;
import com.example.myapplication.model.Transaction;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiService {
    
    @GET("club/{club_pk}/ledger/")
    Call<List<Ledger>> getLedgerList(@Path("club_pk") int clubPk);
    
    @GET("club/{club_pk}/ledger/{id}/")
    Call<Ledger> getLedgerDetail(@Path("club_pk") int clubPk, @Path("id") int ledgerId);
    
    @GET("club/{club_pk}/ledger/{ledger_pk}/transactions/")
    Call<List<Transaction>> getTransactionList(@Path("club_pk") int clubPk, @Path("ledger_pk") int ledgerPk);
    
}

File: MyApplication/MyApplication/app/src/main/java/com/example/myapplication/api/ApiService.kt
package com.example.myapplication.api

import com.example.myapplication.LedgerApiItem
import com.example.myapplication.ClubItem
import com.example.myapplication.TransactionItem
import com.example.myapplication.model.Ledger
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiService {
    @GET("club/{club_pk}/ledger/")
    fun getLedgerList(@Path("club_pk") clubId: Int): Call<List<LedgerApiItem>>
    
    @GET("club/{club_pk}/ledger/{ledger_pk}/")
    fun getLedgerDetail(@Path("club_pk") clubId: Int, @Path("ledger_pk") ledgerId: Int): Call<Ledger>

    @GET("club/")
    fun getClubList(): Call<List<ClubItem>>
    
    @GET("club/{club_pk}/ledger/{ledger_pk}/transactions/")
    fun getTransactions(
        @Path("club_pk") clubId: Int,
        @Path("ledger_pk") ledgerId: Int
    ): Call<List<TransactionItem>>
}
